### ISSUE TITLE
[cmake,2022.10] CopyHeaders does not work as expected

### ISSUE THREAD

**Issue by blattms** [OPM/opm-common#3382]
There is serious error for OPM in Debian, see their [bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1029440) Sometimes the build fails because python/cxx/builtin_pybind11.cpp is not there. Either it is not generated for whatever reason or deleted. (The generator code itself does not do any kind of error checking on file io)

There are some suspicious lines the log:
```
Deleting primary custom command output "/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/ParserKeywords/A.cpp" because \
another output "/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/python/cxx/builtin_pybind11.cpp" does not exist.
```

What seems to happen
- For whatever reason, builtin_pybind11.cpp is not generated.
- Cmake deletes A.cpp (the first generated file) to make that clear
- When the files are copied, this is recognized and the generation is started again. Seems to fail again but is never noticed.
- Compilation fails because of missing builtin_pybind11.cpp.

I have access to a machine to reproduce this.

**Comment by blattms** [comment]
I found the problem: Missing dependencies between targets. With parallel builds we were lucky.

**Comment by blattms** [comment]
Already fixed in master by #3238
